IOPool/Streamer/src/StreamerOutputModuleBase.cc

```diff
@@ -54,7 +54,13 @@ namespace edm::streamer {
   void StreamerOutputModuleBase::write(EventForOutput const& e) {
     Handle<TriggerResults> const& triggerResults = getTriggerResults(trToken_, e);
 
-    std::unique_ptr<EventMsgBuilder> msg = serializeEvent(*getSerializerBuffer(), e, triggerResults, selectorConfig());
+    if (lastCallWasBeginRun_) {
+      auto msg = serializeEventMetaData(*getSerializerBuffer(), *branchIDLists(), *thinnedAssociationsHelper());
+      doOutputEvent(*msg);
+      lastCallWasBeginRun_ = false;
+    }
```

Comment:
After talking with @makortel we determined not to move the logic because of the behavior if there are no events in a LuminosityBlock. The behavior before this PR is the file would contain only the INIT message. The present implementation of the PR would also only write the INIT message. If we changed this PR to serialize the event meta data in the beginRun, a file with no events would then become one with an INIT message followed by a Event message (which just contains the event meta data).  In order to avoid change of behavior, we will keep the code _as is_.